spec.html

```diff
@@ -69,11 +69,12 @@ <h1>Uint8Array.fromBase64 ( _string_ [ , _options_ ] )</h1>
   </emu-alg>
 </emu-clause>
 
-<emu-clause id="sec-uint8array.frombase64into">
-  <h1>Uint8Array.fromBase64Into ( _string_, _into_ [ , _options_ ] )</h1>
+<emu-clause id="sec-uint8array.prototype.setfrombase64">
+  <h1>Uint8Array.prototype.setFromBase64 ( _string_ [ , _options_ ] )</h1>
   <emu-alg>
-    1. If _string_ is not a String, throw a *TypeError* exception.
+    1. Let _into_ be the *this* value.
```

Comment:
```suggestion
    1. Let _into_ be the *this* value.
    1. Perform ? ValidateTypedArray(_into_, ~seq-cst~).
```

Here and below to match other TypedArray.prototype built-ins.

Reply:
The closest such built-in is [TypedArray.prototype.set](https://tc39.es/ecma262/multipage/indexed-collections.html#sec-%typedarray%.prototype.set), and it, like this method, initially checks only the presence of the internal slot on `this`, and then does the potentially side-effecting handling of arguments, and [only after that](https://tc39.es/ecma262/multipage/indexed-collections.html#sec-settypedarrayfromtypedarray) does the `IsTypedArrayOutOfBounds` check.

That's what I'm doing here.

Reply:
`TypedArray.prototype.set` (and `TypedArray.prototype.subarray`) are both legacy Khronos-based functions, so I don't think it's useful/necessary to copy the same approach in new methods. Instead it's preferable to use the same validation sequence which is present in all newer TypedArray built-ins.

Reply:
(Actually it'd great if we could change `TypedArray.prototype.{set,subarray}` to call `ValidateTypedArray` at the start, because it's kind of annoying that `subarray` can observe the original `[[ByteOffset]]` when the underlying buffer is detached, see <https://bugzilla.mozilla.org/show_bug.cgi?id=1291003> and <https://bugzilla.mozilla.org/show_bug.cgi?id=1840991>. Being able to read `[[ByteOffset]]` from detached/out-of-bounds typed arrays also requires extra JIT support in SpiderMonkey, where `subarray` is a self-hosted built-in: <https://phabricator.services.mozilla.com/D200658>.)

Reply:
You want to get the length (and validate non-detachedness) after side-effects happen anyway. Checking up-front would mean duplicating that work.

Reply:
cc @syg for thoughts.

Reply:
> You want to get the length (and validate non-detachedness) after side-effects happen anyway. Checking up-front would mean duplicating that work.

I agree with this.

I'm leaning towards breaking with consistency. If we do that, the most desirable argument coercion order for TypedArray builtins are to first process all the parameters, _then_ do a `ValidateTypedArray` on **this**. After that point, with the exception of built-ins that call user-provided callbacks in a loop (like `map` or something), there should be no more user code being called.

If for whatever reason committee is not amenable to breaking with consistency, the `ValidateTypedArray` on **this**, then arguments, then `ValidateTypedArray` again order is wasteful. And given that maximal consistency is not possible since the builtins are already inconsistent, I see no compelling reason to be wasteful.